### PR TITLE
Implemented SCRAM-SHA1 authentication method for mongo

### PR DESCRIPTION
**Pull request by ngurrola**
The default is still MONGO-CR, even for server versions 3.0+ to maintain backwards compatibility. It can be switched by adding "authMechanism=SCRAM-SHA-1" to the connection query string.

### CODE REVIEW

File: mongodb/vibe/db/mongo/sasl.d [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+import std.digest.sha;
+import std.exception;
+import std.format;
+import std.string;
+import std.utf;
+import vibe.crypto.cryptorand;
+
+private SHA1HashMixerRNG g_rng;
+
+struct ScramState
```
Comment (s-ludwig) [inline mongodb/vibe/db/mongo/sasl.d]: I'd mark this as `package`, so that we don't have to worry about the API so much for now.

**Comment by s-ludwig** [comment]
Really great to finally have this! I never had the time (and need) to implement it, even though it's a rather important feature.

**Comment by ngurrola** [comment]
I added those changes as requested.

**Comment by s-ludwig** [comment]
Looks good to merge. Thanks a lot!

**Pull request merged**